Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM python:alpine as base
+
+EXPOSE 8000
+RUN apk add --no-cache \
+    curl \
+    npm \
+    nodejs
+RUN curl -sSL https://install.python-poetry.org | python3 -
+ENV PATH="/root/.local/bin:$PATH"
+
+WORKDIR /code
+COPY /package.json /code
+COPY /package-lock.json /code
+COPY /poetry.toml /code
+COPY /poetry.lock /code
+COPY /pyproject.toml /code
+COPY /tailwind.config.js /code
+RUN poetry install
+RUN npm install
+
+FROM base as production
+
+COPY /todo_app /code/todo_app
+RUN npx tailwindcss -i ./todo_app/tailwind.css -o ./todo_app/static/css/index.css
+
+CMD poetry run gunicorn -w=4 --bind 0.0.0.0 "todo_app.app:create_app()"
```

Comment:
I'm a fan of defaulting to ENTRYPOINT over CMD, for example [for the reasons given here](https://spacelift.io/blog/docker-entrypoint-vs-cmd#the-difference-between-docker-entrypoint-and-cmd):
> Incorrectly setting CMD instead of ENTRYPOINT is a common mistake. While your image will usually still work—because Docker defaults to using /bin/sh -c as the ENTRYPOINT—end users won’t be able to directly pass arguments to your binary using docker run. They’ll need to pass the full path to the binary instead, as the container’s ENTRYPOINT process will be /bin/sh -c instead of your own app.

Although I would concede it's a pretty weak statement, and if you can give any reason to disagree it's not something I'd argue with!

Reply:
This makes sense to me... I was a bit unclear on the differences tbh 😅